include/sys/btree.h

```diff
@@ -115,7 +120,44 @@ typedef struct btree {
 	uint64_t		bt_num_nodes;
 	zfs_btree_hdr_t		*bt_root;
 	zfs_btree_leaf_t	*bt_bulk; // non-null if bulk loading
-} zfs_btree_t;
+};
+
+/*
+ * Implementation of Shar's algorithm designed to accelerate binary search by
+ * eliminating impossible to predict branches. For optimality, this should be
+ * used to generate the search function in the same file as the comparator and
+ * the comparator must be marked `__attribute__((always_inline) inline` so that
+ * the compiler will inline the search function.
+ */
```

Comment:
A minor nit. Arguments should be documented like they are for zfs_btree_create().

Reply:
Fixed.